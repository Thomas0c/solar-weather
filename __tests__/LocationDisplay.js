import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LocationDisplay from '../lib/js/app/components/locationDisplay';

test('renders correctly when not loading', () => {
  const tree = renderer.create(<LocationDisplay locationName="Tokyo" loading={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when loading', () => {
  const tree = renderer.create(<LocationDisplay locationName="Tokyo" loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
