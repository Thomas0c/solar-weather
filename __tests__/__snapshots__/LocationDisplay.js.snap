// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when loading 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "bottom": 0,
      "height": 133.4,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "width": "100.00%",
      "zIndex": 1,
    }
  }
  testID={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "backgroundColor": "#FFF",
        "height": 2,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
        "top": 0,
        "width": "100.00%",
        "zIndex": -1,
      }
    }
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "height": 133.4,
            "justifyContent": "flex-start",
            "position": "relative",
            "width": "100.00%",
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        adjustsFontSizeToFit={undefined}
        allowFontScaling={true}
        ellipsizeMode="tail"
        minimumFontScale={undefined}
        numberOfLines={undefined}
        onLayout={undefined}
        onLongPress={undefined}
        onPress={undefined}
        pressRetentionOffset={undefined}
        selectable={undefined}
        selectionColor={undefined}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "fontFamily": "Baskerville",
            "fontSize": 18,
            "height": "100.00%",
            "justifyContent": "center",
            "paddingTop": 56.7,
            "textAlign": "center",
            "width": "100.00%",
          }
        }
        suppressHighlighting={undefined}
        testID={undefined}
        textBreakStrategy={undefined}
      >
        Loading...
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly when not loading 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "bottom": 0,
      "height": 133.4,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "width": "100.00%",
      "zIndex": 1,
    }
  }
  testID={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "backgroundColor": "#FFF",
        "height": 2,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
        "top": 0,
        "width": "100.00%",
        "zIndex": -1,
      }
    }
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "height": 133.4,
            "justifyContent": "flex-start",
            "position": "relative",
            "width": "100.00%",
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        adjustsFontSizeToFit={undefined}
        allowFontScaling={true}
        ellipsizeMode="tail"
        minimumFontScale={undefined}
        numberOfLines={undefined}
        onLayout={undefined}
        onLongPress={undefined}
        onPress={undefined}
        pressRetentionOffset={undefined}
        selectable={undefined}
        selectionColor={undefined}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "fontFamily": "Baskerville",
            "fontSize": 18,
            "height": "100.00%",
            "justifyContent": "center",
            "paddingTop": 56.7,
            "textAlign": "center",
            "width": "100.00%",
          }
        }
        suppressHighlighting={undefined}
        testID={undefined}
        textBreakStrategy={undefined}
      >
        Tokyo
      </Text>
    </View>
  </View>
</View>
`;
